siatest/testgroup_test.go

```diff
@@ -98,3 +99,45 @@ func TestNewGroupNoRenterHost(t *testing.T) {
 		}
 	}()
 }
+
+// TestAddNewNode tests that the added node is returned when AddNodes is called
+func TestAddNewNode(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	// Create a group
+	groupParams := GroupParams{
+		Renters: 2,
+		Miners:  1,
+	}
+	tg, err := NewGroupFromTemplate(groupParams)
+	if err != nil {
+		t.Fatal("Failed to create group: ", err)
+	}
+	defer func() {
+		if err := tg.Close(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// Record current nodes
+	oldRenters := tg.Renters()
+
+	// Test adding a node
+	testDir, err := TestDir(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	renterTemplate := node.Renter(testDir + "/renter")
+	nodes, err := tg.AddNodes(renterTemplate)
```

Comment:
You should also check the length of `nodes` since it might have a different length than `1`.